connectors/sync_job_runner.py

```diff
@@ -0,0 +1,175 @@
+
+        try:
+            self.data_provider = self.source_klass(self.sync_job.configuration)
+            if not await self.data_provider.changed():
+                logger.debug(
+                    f"No change in {self.sync_job.service_type} data provider, skipping..."
+                )
+                await self._sync_done(sync_status=JobStatus.COMPLETED, result={})
+                return
+
+            logger.debug(f"Validating configuration for {self.data_provider}")
+            await self.data_provider.validate_config()
+
+            logger.debug(
+                f"Syncing '{self.sync_job.service_type}' for connector '{self.connector_id}'"
+            )
+            logger.debug(f"Pinging the {self.source_klass} backend")
+            await self.data_provider.ping()
+
+            sync_rules_enabled = self.connector.features.sync_rules_enabled()
+            if sync_rules_enabled:
+                await self.sync_job.validate_filtering(validator=self.data_provider)
+
+            mappings = Mappings.default_text_fields_mappings(
+                is_connectors_index=True,
+            )
+
+            logger.debug("Preparing the content index")
+            await self.elastic_server.prepare_content_index(
+                self.sync_job.index_name, mappings=mappings
+            )
+
+            # allows the data provider to change the bulk options
+            bulk_options = self.bulk_options.copy()
+            self.data_provider.tweak_bulk_options(bulk_options)
+
+            result = await self.elastic_server.async_bulk(
+                self.sync_job.index_name,
+                self.prepare_docs(),
+                self.sync_job.pipeline,
+                filter_=self.sync_job.filtering,
+                sync_rules_enabled=sync_rules_enabled,
+                options=bulk_options,
+            )
+            sync_error = result.get("fetch_error")
+            sync_status = JobStatus.COMPLETED if sync_error is None else JobStatus.ERROR
+            await self._sync_done(
+                sync_status=sync_status, result=result, sync_error=sync_error
+            )
+        except asyncio.CancelledError:
+            await self._sync_done(sync_status=JobStatus.SUSPENDED, result={})
+        except Exception as e:
+            await self._sync_done(sync_status=JobStatus.ERROR, result={}, sync_error=e)
+        finally:
+            self._start_time = None
+            if self.data_provider is not None:
+                await self.data_provider.close()
+
+    async def _sync_done(self, sync_status, result=None, sync_error=None):
+        if result is None:
+            result = {}
+        doc_updated = result.get("doc_updated", 0)
+        doc_created = result.get("doc_created", 0)
+        doc_deleted = result.get("doc_deleted", 0)
+        indexed_count = doc_updated + doc_created
+
+        ingestion_stats = {
+            "indexed_document_count": indexed_count,
+            "indexed_document_volume": 0,
```

Comment:
Should this be zero or do we need to calculate something here?

Reply:
This will be addressed later: https://github.com/elastic/connectors-python/pull/515